schema/swagger-extensions.json

```diff
@@ -1122,6 +1142,9 @@
         "example": { },
         "x-ms-client-flatten": {
           "$ref": "#/definitions/xmsClientFlatten"
+        },
+        "x-ms-client-name": {
+          "type": "string"
```

Comment:
I think we need to be careful when describing extensions with a root of anything other than Object. When we define extensions as primitives (bool, string, number), then we lock ourselves into that structure. The soft nature of objects and keys will allow for extensibility and adaptation later as the extension matures.

Thoughts?